Admin Util force stop error

If you try to force stop a server that is not running using the `accumulo admin` command, an error will be thrown:
<pre>
09:55:47 {main} ~/workspace/uno$ accumulo admin -f stop localhost:9997 
2022-05-31T09:55:54,195 [conf.SiteConfiguration] INFO : Found Accumulo configuration on classpath at /home/mpmill4/workspace/uno/install/accumulo-2.1.0-SNAPSHOT/conf/accumulo.properties
2022-05-31T09:55:55,062 [metrics.MetricsUtil] INFO : Metric producer PropStoreMetrics initialize
2022-05-31T09:55:55,197 [util.Admin] INFO : Stopping server localhost:9997
2022-05-31T09:55:55,245 [util.Admin] ERROR: org.apache.thrift.TApplicationException: Internal error processing shutdownTabletServer
org.apache.accumulo.core.client.AccumuloException: org.apache.thrift.TApplicationException: Internal error processing shutdownTabletServer
	at org.apache.accumulo.core.rpc.clients.ManagerThriftClient.executeVoidTableCommand(ManagerThriftClient.java:123) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
	at org.apache.accumulo.core.rpc.clients.ManagerThriftClient.executeVoid(ManagerThriftClient.java:135) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
	at org.apache.accumulo.server.util.Admin.stopTabletServer(Admin.java:398) ~[accumulo-server-base-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
	at org.apache.accumulo.server.util.Admin.execute(Admin.java:261) ~[accumulo-server-base-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
	at org.apache.accumulo.start.Main.lambda$execKeyword$0(Main.java:126) ~[accumulo-start-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:829) ~[?:?]
Caused by: org.apache.thrift.TApplicationException: Internal error processing shutdownTabletServer
	at org.apache.thrift.TServiceClient.receiveBase(TServiceClient.java:79) ~[libthrift-0.16.0.jar:0.16.0]
	at org.apache.accumulo.core.manager.thrift.ManagerClientService$Client.recv_shutdownTabletServer(ManagerClientService.java:404) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
	at org.apache.accumulo.core.manager.thrift.ManagerClientService$Client.shutdownTabletServer(ManagerClientService.java:388) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
	at org.apache.accumulo.server.util.Admin.lambda$stopTabletServer$2(Admin.java:399) ~[accumulo-server-base-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
	at org.apache.accumulo.core.rpc.clients.ManagerThriftClient.executeVoidTableCommand(ManagerThriftClient.java:102) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
	... 5 more
</pre>

But if you don't use the `-f, --force` option, nothing happens. The error should not be happening. If it is possible, we should also log that the server is not running.

--- a/server/manager/src/main/java/org/apache/accumulo/manager/ManagerClientServiceHandler.java
+++ b/server/manager/src/main/java/org/apache/accumulo/manager/ManagerClientServiceHandler.java
@@ -261,10 +261,14 @@ public class ManagerClientServiceHandler implements ManagerClientService.Iface {
       throw new ThriftSecurityException(c.getPrincipal(), SecurityErrorCode.PERMISSION_DENIED);
 
     final TServerInstance doomed = manager.tserverSet.find(tabletServer);
+    if (doomed == null) {
+      Manager.log.warn("No server found for name {}, unable to shut it down", tabletServer);
+      return;
+    }
     if (!force) {
       final TServerConnection server = manager.tserverSet.getConnection(doomed);
       if (server == null) {
-        Manager.log.warn("No server found for name {}", tabletServer);
+        Manager.log.warn("No server found for name {}, unable to shut it down", tabletServer);
         return;
       }
     }

--- a/test/src/main/java/org/apache/accumulo/test/functional/ManagerApiIT.java
+++ b/test/src/main/java/org/apache/accumulo/test/functional/ManagerApiIT.java
@@ -313,6 +313,20 @@ public class ManagerApiIT extends SharedMiniClusterBase {
     expectPermissionSuccess(op, privilegedUser);
   }
 
+  @Test
+  public void shutdownTabletServer() throws Exception {
+    op = client -> {
+      client.shutdownTabletServer(TraceUtil.traceInfo(), rootUser.toThrift(instanceId),
+          "fakeTabletServer:9997", true);
+      return null;
+    };
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps())
+        .as(rootUser.getPrincipal(), rootUser.getToken()).build()) {
+      ClientContext context = (ClientContext) client;
+      ThriftClientTypes.MANAGER.execute(context, op);
+    }
+  }
+
   // this test should go last, because it shuts things down;
   // see the junit annotation to control test ordering at the top of this class
   @Test

